### DISCUSSION TITLE
snowflake-labs provider upgrade

### DISCUSSION THREAD

**Discussion by tabu69** [Snowflake-Labs/terraform-provider-snowflake#2850] [Q&A] [answered]
Hi Community,

I have a use case for dynamic tables and note the earliest provider version this functionality is available in, is 0.75.0, current provider version being used in my code is 0.48.0 (yes, super old!), when testing upgrade of version, the following error is outputted.

Error: Provider produced invalid plan
Provider "registry.terraform.io/snowflake-labs/snowflake" planned an invalid
value for
snowflake_role_grants.set_role_to_role_grants_own_to_sysadmin.users:
planned value cty.SetValEmpty(cty.String) for a non-computed attribute.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.

Example of the code for this output:

## CREATE OWN ROLE ##
resource "snowflake_role" "OWN" {
  name    = "OWN"
  comment = "Role for sysadmin"
}

resource "snowflake_role_grants" "set_role_to_role_grants_own_to_sysadmin" {
  role_name = snowflake_role.OWN.name
  roles     = ["SYSADMIN"]
}

Also seeing strange outputs for warehouse

------------------------------------
Error: Provider produced invalid plan

Provider "registry.terraform.io/snowflake-labs/snowflake" planned an invalid
value for
snowflake_warehouse.warehouse["TEST_P01_WH"].query_acceleration_max_scale_factor:
planned value cty.NumberIntVal(8) for a non-computed attribute.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

Error: Provider produced invalid plan

Provider "registry.terraform.io/snowflake-labs/snowflake" planned an invalid
value for
snowflake_warehouse.warehouse["TEST_P01_WH"].statement_queued_timeout_in_seconds:
planned value cty.NumberIntVal(0) for a non-computed attribute.

------------------------------------

I have tested with later provider versions (example 0.88.0, 0.90.0) and the error is the same, even when using new resource types for role grant. The same code structure works perfectly fine in provider version 0.48.0

I am wonder if a sensible next step would be to upgrade terraform version? I am currently using version 1.0.8, which is old

Would be grateful for any advice from community on this please.

Thanks!

**Comment by sfc-gh-jcieslak** [comment, marked as answer]
Hey @tabu69 
Great to hear that you're trying to keep the provider updated. The error you provided seems like a common error that comes from the old Terraform CLI version. Please check [this doc](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/CREATING_ISSUES.md#old-terraform-cli-version) where we described this error and other common issues. Let me know if upgrading to at least Terraform CLI 1.1.5 resolves the error.

**Reply by tabu69** [reply]
Hey @sfc-gh-jcieslak  - Thank you for the quick response and sharing details. I will test with upgraded terraform version and see how it goes; will update on outcome. Thanks again.

**Reply by tabu69** [reply]
Hey @sfc-gh-jcieslak, upgrade of the terraform version did the trick! Thanks very much for the guidance :)
